### DISCUSSION TITLE
Add ability to add file information within store for monitoring purposes

### DISCUSSION THREAD

**Discussion by alexis-bvq** [atoti/atoti#639] [General]
## Description
Hello, I am upgrading a java project to a python project. In this project, they load new files everyday and in order to know which data came from where they have column calculators adding the filename and the filedate within each store. It is definitely something you don't want to add in the original file as it is intrinsic data attached to each file and do not need to be replicated over each rows.
It would be very nice to have an option allowing to add those information within the store when you load data.

**Comment by alexis-bvq** [comment]
The implementation of this project will start in about a month, it could be good to keep that in mind as the client might/will request it to have the results as they had with their java implementation.

**Comment by Acriss** [comment]
>  It would be very nice to have an option allowing to add those information within the store when you load data.

I don't exactly understand what you're suggesting.
Do you suggest that we could add have something like `table.read_csv(store_file_name=true, store_loading_date_and_time=true)`, which would add 2 column calculators that enrich each line of the store with the name of the file it comes from, and the date/time it was loaded into the store ?

Note: Performance wise, this shouldn't be an issue. Loading time might get a bit longer, but not by much.
Memory wise, this shouldn't be an issue either, as we'll likely load quite a few lines per file, so that the chunks of data will be compressed into `SingleValueChunks`.

**Comment by alexis-bvq** [comment]
Yes this is exactly what would be required for this feature :)

**Comment by rcolle** [comment]
> Do you suggest that we could add have something like table.read_csv(store_file_name=true, store_loading_date_and_time=true), which would add 2 column calculators that enrich each line of the store with the name of the file it comes from, and the date/time it was loaded into the store ?

I think I'd prefer an `extra_columns` parameter that would take a column name to value mapping like `{"file_name": "myFile.csv", "loading_date": datetime.now()}`

**Comment by alexis-bvq** [comment]
In case you have a watcher on a folder where there is a drop of files, you would have to notice the drop of files and could do read_csv("folder/**.csv"), but here with your solution rcolle we would have to start a transaction, read all the file name by name in order to add the value in the mapping. Knowing that when you start a transaction you block the session of doing anything else it can be annoying.
